graphics/opengl/context.cpp

```diff
@@ -178,39 +246,58 @@ void ContextGL::initialize(ContextOGLType contextType) {
 		if (framebufferObjectMultisampleSupported) {
 			glGetIntegerv(GL_MAX_SAMPLES, (GLint *)&multisampleMaxSamples);
 		}
-	}
 
-	// Log context type.
-	switch (type) {
-		case kOGLContextNone:
-			/* Shouldn't happen */
-			break;
-		case kOGLContextGL:
-			debug(5, "OpenGL: GL context initialized");
-			break;
-
-		case kOGLContextGLES2:
-			debug(5, "OpenGL: GLES2 context initialized");
-			break;
+		// OpenGL 1.2 and later always has packed pixels and texture edge clamp support
+		if (isGLVersionOrHigher(1, 2)) {
+			packedPixelsSupported = true;
+			textureEdgeClampSupported = true;
+		}
+		debug(5, "OpenGL: GL context initialized");
+	} else {
+		warning("OpenGL: Unknown context initialized");
 	}
 
 	// Log features supported by GL context.
+	debug(5, "OpenGL vendor: %s", glGetString(GL_VENDOR));
+	debug(5, "OpenGL renderer: %s", glGetString(GL_RENDERER));
+	debug(5, "OpenGL: version %d.%d", majorVersion, minorVersion);
+	debug(5, "OpenGL: GLSL version: %d", glslVersion);
+	debug(5, "OpenGL: Max texture size: %d", maxTextureSize);
 	debug(5, "OpenGL: NPOT texture support: %d", NPOTSupported);
 	debug(5, "OpenGL: Shader support: %d", shadersSupported);
+	debug(5, "OpenGL: Multitexture support: %d", multitextureSupported);
 	debug(5, "OpenGL: FBO support: %d", framebufferObjectSupported);
+	debug(5, "OpenGL: Multisample FBO support: %d", framebufferObjectMultisampleSupported);
+	debug(5, "OpenGL: Multisample max number: %d", multisampleMaxSamples);
+	debug(5, "OpenGL: Packed pixels support: %d", packedPixelsSupported);
 	debug(5, "OpenGL: Packed depth stencil support: %d", packedDepthStencilSupported);
 	debug(5, "OpenGL: Unpack subimage support: %d", unpackSubImageSupported);
+	debug(5, "OpenGL: OpenGL ES depth 24 support: %d", OESDepth24);
+	debug(5, "OpenGL: Texture edge clamping support: %d", textureEdgeClampSupported);
 }
 
-int ContextGL::getGLSLVersion() const {
+int Context::getGLSLVersion() const {
+#if USE_FORCED_GLES
+	return 0;
+#else
+	// No shader support in GLES
+	if (type == kContextGLES) {
+		return 0;
+	}
+
+	// No shader support in OpenGL 1.x
+	if (type == kContextGL && !isGLVersionOrHigher(2, 0)) {
+		return 0;
+	}
+
```

Comment:
that is true, but do we want to add it? also does GLAD do aliasing for ARB extension to make it as core? if not adding this will require also add support shaders for extension. Personal I don't see benefit supporting this case. I think this is not relevant in current days. Fo shaders supported cards through ARB, they will also support later GL version.

Reply:
Aliasing is done by GLAD but if we do something it will be only for this case.

Reply:
After looking up at the backend code, it seems we tried to support ARB specifically at some point but not anymore: the only leftover is `GLprogram` and `GLshader` typedefs.
I propose to rely on GLAD aliasing only in this case.
I added a documentation to this function: it only takes care of native GLSL version.
ARB case is handled above.

Reply:
ok, fine